tests/framework/installer/ceph_installer.go

```diff
@@ -235,6 +265,143 @@ func (h *CephInstaller) CreateK8sRookClusterWithHostPathAndDevices(namespace, sy
+	}
+
+	// Skip this step since the helm chart already includes the roles and bindings
+	if !h.useHelm {
+		logger.Infof("Creating external cluster roles")
+		roles := h.Manifests.GetClusterExternalRoles(namespace, firstClusterNamespace)
+		if _, err := h.k8shelper.KubectlWithStdin(roles, createFromStdinArgs...); err != nil {
+			return fmt.Errorf("failed to create cluster roles. %v", err)
+		}
+	}
+
+	// Inject connection information from the first cluster
+	logger.Info("Injecting cluster connection information")
+	err = h.InjectRookExternalClusterInfo(namespace, firstClusterNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to inject cluster information into the external cluster. %v", err)
+	}
+
+	// Start the external cluster
+	logger.Infof("Starting Rook External Cluster with yaml")
+	settings := &ClusterExternalSettings{namespace, dataDirHostPath}
+	rookCluster := h.Manifests.GetRookExternalCluster(settings)
+	if _, err := h.k8shelper.KubectlWithStdin(rookCluster, createFromStdinArgs...); err != nil {
+		return fmt.Errorf("failed to create rook external cluster. %v ", err)
+	}
+
+	logger.Infof("Rook external cluster started")
+	return err
+}
+
+// InjectRookExternalClusterInfo inject connection information for an external cluster
+func (h *CephInstaller) InjectRookExternalClusterInfo(namespace, firstClusterNamespace string) error {
+	args := []string{"apply", "--namespace", namespace, "-f", "-"}
+
+	// get cm
+	cm, err := h.GetRookExternalClusterMonConfigMap(firstClusterNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to get secret. %v", err)
+	}
+
+	// get secret
+	secret, err := h.GetRookExternalClusterMonSecret(firstClusterNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to get configmap. %v", err)
+	}
+
+	// create cm
+	_, err = h.k8shelper.KubectlWithStdin(cm, args...)
+	if err != nil {
+		return fmt.Errorf("failed to create cm. %v", err)
+	}
+
+	// create secret
+	_, err = h.k8shelper.KubectlWithStdin(secret, args...)
+	if err != nil {
+		return fmt.Errorf("failed to create secret. %v", err)
+	}
+
+	return nil
+}
+
+// GetRookExternalClusterMonConfigMap gets the monitor kubernetes configmap of the external cluster
+func (h *CephInstaller) GetRookExternalClusterMonConfigMap(namespace string) (string, error) {
+	configMapName := "rook-ceph-mon-endpoints"
+	args := []string{"--namespace", namespace, "get", "configmap", configMapName, "-o", "json"}
+
+	cm, err := h.k8shelper.Kubectl(args...)
+	if err != nil {
+		return "", fmt.Errorf("failed to get kubernetes monitor configmap endpoint %q. %v", configMapName, err)
+	}
+
+	// Since the --export is being deprecated: https://github.com/kubernetes/kubernetes/pull/73787
+	// We need to process the output and modify it a bit
+	// We remove the
+	// 'namespace'
+	// 'creationTimestamp'
+	// 'resourceVersion'
+	// 'uid'
+	cmByte := []byte(cm)
+	var m KubernetesConfigMap
+	if err := json.Unmarshal([]byte(cmByte), &m); err != nil {
```

Comment:
Instead of getting the exported yaml and deserializing, how about calling the k8s API to get the value of the configmap then create a new configmap with its value. Something like this:
```
externalCM, err := clientset.CoreV1().ConfigMaps(namespace).Get(name)
newCM := &v1.ConfigMap{}
newCM.Data = externalCM.Data
clientset.CoreV1().ConfigMaps(namespace).Create(newCM)
```

Reply:
Way better!